src/battleSystem/hpGauge.mjs

```diff
@@ -126,76 +350,604 @@ class HPGauge
+		this._shader = shader;
+
+		//state
+		this._state = full;
+		
+		//housekeeping
+		this._needsRender = true;
+		this._needsTranslate = true;
+		this._needsModel = true;
+    }
+
+    reRender ()
+    {
+		this._full = renderFilled(this._width - 2, this._height - 2, white);
+		this._damaged = renderFilled(this._width - 2, this._height - 2, this._damageColour);
+		this._empty = renderFilled(this._width - 2, this._height - 2, this._emptyColour);
+		this._outer = renderOutline(this._width, this._height, 1, this._borderColour);
+		this._needsModel = true;
+        this._needsRender = false;
+	}
+
+	model ()
+	{
+		switch (this._state)
+		{
+			case (full):
+				this._model = new Model([this._outer, this._full]);
+				this._model.shader = this._hpShader;
+				break;
+			case (damaged):
+				this._model = new Model([this._outer, this._damaged]);
+				this._model.shader = this._shader;
+				break;
+			case (empty):
+				this._model = new Model([this._outer, this._empty]);
+				this._model.shader = this._shader;
+				break;
+		}
+		this._model.transform = this._transform;
+		this._needsModel = false;
+	}
+	
+	translate ()
+	{
+		this._transform.identity();
+		this._transform.translate(this._x, this._y);
+		this._needsTranslate = false;
+	}
+
+	set state (value)
+	{
+		if (this._state !== value)
+		{
+			this._state = value;
+			this._needsModel = true;
+		}
+	}
+
+	get x () { return this._x; };
+	get y () { return this._y; };
+
+	set parentX (value)
+	{
+		let difference = value - this._parentX;
+		if (difference !== 0)
+		{
+			this._parentX = value;
+			this.x = this._x + difference;
+		}
+	}
+
+	set parentY (value)
+	{
+		let difference = value - this._parentY;
+		if (difference !== 0)
+		{
+			this._parentY = value;
+			this.y = this._y + difference;
+		}
+	}
+
+	set childY (value)
+	{
+		let difference = value - this._childY;
+		if (difference !== 0)
+		{
+			this._childY = value;
+			this.y = this._y + difference;
+		}
+	}
+
+	set childX (value)
+	{
+		let difference = value - this._childX;
+		if (difference !== 0)
+		{
+			this._childX = value;
+			this.x = this._x + difference;
+		}
+	}
+
+    set y (value)
+    {
+        if (this._y !== value)
+        {
+            this._y = value;
+            this._needsTranslate = true;
+        }
+    }
+
+    set x (value)
+    {
+        if (this._x !== value)
+        {
+            this._x = value;
+            this._needsTranslate = true;
+        }
+	}
+	
+	set width (value)
+	{
+		if(this._width !== value)
+		{
+			this._width = value;
+			this._needsRender = true;
+		}
+	}
+
+	set height (value)
+	{
+		if(this._height !== value)
+		{
+			this._height = value;
+			this._needsRender = true;
+		}
+	}
+
+    draw ()
+    {
+        if (this._needsRender === true)
+        {
+            this.reRender();
+		}
+		if (this._needsTranslate === true)
+		{
+			this.translate();
+		}
+		if (this._needsModel == true)
+		{
+			this.model();
+		}
+        this._model.draw(Surface.Screen);
```

Comment:
API documentation says `Model#draw()` surface argument is optional - is that not the case in practice?

Reply:
mistake - will remove